src/main/java/me/danjono/inventoryrollback/listeners/ClickGUI.java

```diff
@@ -355,7 +361,19 @@ else if (icon.getType().equals(Buttons.getTeleportLocationIcon())) {
                 }
 
                 String[] location = nbt.getString("location").split(",");			
-                World world = Bukkit.getWorld(location[0]);
+
+                if (!Objects.equals(ConfigData.getServerName(), location[0])) {
+                    ByteArrayOutputStream b = new ByteArrayOutputStream();
+                    DataOutputStream out = new DataOutputStream(b);
+
+                    out.writeUTF("Connect");
+                    out.writeUTF(location[0]);
+
+                    staff.sendPluginMessage(InventoryRollback.getInstance(), "BungeeCord", b.toByteArray());
+                    return;
+                }
```

Comment:
Util class here would be nice

Reply:
Also, no need for redis to teleport to the correct location. Setting an expiring connection teleport 'intent' (or whatever you want to call it), would be sufficient: user click TP -> intent entry is recorded into DB table (await) -> Bungee msg -> user async pre-login -> lookup (with SQL time validity check condition) -> hold intent in cache -> user joins (PlayerSpawnLocationEvent iirc) -> execute location set (effectively teleport) if location valid for the server & permissions for TP still exist.

User disconnection or login failure -> cache invalidation